### ISSUE TITLE
[bug] Unable to use pre-installed msys2 subsystem to build packages

### ISSUE THREAD

**Issue by wbehrens-on-gh** [conan-io/conan#15313]
### Environment details

* Operating System+version: Windows 11
* Compiler+version: clang 8
* Conan version: 2.0.14
* Python version: 3.8.10

### Steps to reproduce

I'm unsure if this is a conan issue or an issue with the conan-center-index openssl recipe. Currently I'm attempting to cross build openssl for android on a Windows 11 machine. I had been running into issues where our CI pipeline would hang when conan attempts to install the msys2 package so I thought to get around that issue by pre-installing it on the system and having the `global.conf` reference it.

build profile
```
[settings]
os=Windows
arch=x86_64
compiler=msvc
compiler.version=192
compiler.runtime=dynamic
build_type=Release
```

host profile
```
{% set api_level = 16 %}
{% set target_host = 'arm-linux-androideabi' %}
{% set clang_host = 'armv7a-linux-androideabi' %}
{% set android_ndk = os.getenv("ANDROID_NDK") %}

[settings]
os=Android
os.api_level={{api_level}}
arch=armv7
compiler=clang
compiler.libcxx=c++_shared
compiler.version=8
build_type=Release

[conf]
# In newer NDK versions (>=21) should only require tools.android:ndk_path to be set
tools.android:ndk_path={{android_ndk}}
tools.gnu:host_triplet={{target_host}}
tools.build:compiler_executables={'c': '{{clang_host}}{{api_level}}-clang', 'cpp': '{{clang_host}}{{api_level}}-clang++'}

[buildenv]
PATH+=(path){{ os.path.join(android_ndk, 'toolchains', 'llvm', 'prebuilt', 'windows-x86_64', 'bin') }}
```

global.conf
```
# Force the installation of required items
tools.system.package_manager:mode=install

# Tell Windows where to find bash 
tools.microsoft.bash:path=C:\msys2\usr\bin\bash.exe
tools.microsoft.bash:subsystem=msys2
```

### Logs

```
-------- Installing package openssl/1.1.1t (3 of 4) --------
openssl/1.1.1t: Building from source
openssl/1.1.1t: Package openssl/1.1.1t:60ec2ce269923046ab34580c8701f1000b99d70f
openssl/1.1.1t: Copying sources to build folder
openssl/1.1.1t: Building your package in C:\Conan\p\b\opens7f7f708f2435a\b
openssl/1.1.1t: Calling generate()
openssl/1.1.1t: Generators folder: C:\Conan\p\b\opens7f7f708f2435a\b\build-debug\conan
openssl/1.1.1t: Generating aggregated env files
openssl/1.1.1t: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
openssl/1.1.1t: Calling build()
openssl/1.1.1t: Apply patch (portability): TVOS and WatchOS don't like fork()
openssl/1.1.1t: gen_info = {'CFLAGS': ['-fPIC', '-g'], 'CXXFLAGS': ['-fPIC', '-g'], 'DEFINES': [], 'LDFLAGS': []}
openssl/1.1.1t: using target: conan-Debug-Android-armv7-clang-8 -> linux-generic32
openssl/1.1.1t: my %targets = (
    "conan-Debug-Android-armv7-clang-8" => {
        inherit_from => [ "linux-generic32" ],
        cflags => add("-fPIC -g"),
        cxxflags => add("-fPIC -g"),

        includes => add(),
        lflags => add(""),







        perlasm_scheme => "void",
    },
);

openssl/1.1.1t: ['"conan-Debug-Android-armv7-clang-8"', 'no-shared', '--prefix=/', '--openssldir="res"', 'no-unit-test', 'threads', 'PERL=perl', 'no-tests', '--debug', '--libdir=lib', ' -D__ANDROID_API__=16', '-fPIC', 'no-md2']
openssl/1.1.1t: RUN: perl ./Configure "conan-Debug-Android-armv7-clang-8" no-shared --prefix=/ --openssldir="res" no-unit-test threads PERL=perl no-tests --debug --libdir=lib  -D__ANDROID_API__=16 -fPIC no-md2
The system cannot find the path specified.

openssl/1.1.1t: ERROR:
Package '60ec2ce269923046ab34580c8701f1000b99d70f' build failed
openssl/1.1.1t: WARN: Build folder C:\Conan\p\b\opens7f7f708f2435a\b\build-debug
*********************************************************
Recipe 'openssl/1.1.1t' cannot build its binary
It is possible that this recipe is not Conan 2.0 ready
If the recipe comes from ConanCenter, report it at https://github.com/conan-io/conan-center-index/issues
If it is your recipe, check if it is updated to 2.0
*********************************************************

ERROR: openssl/1.1.1t: Error in build() method, line 723
        self.run(f'{self._perl} ./Configure {args}')
        ConanException: Error 1 while executing
```
It doesn't seem to give any hints as to what wasn't found?

**Comment by memsharded** [comment]
Quick question. Does ``openssl/1.1.1`` cross-build correctly to Android if using the ConanCenter ``msys2`` package instead of your own, with exactly same profile (minus the specific msys2 details)?

**Comment by wbehrens-on-gh** [comment]
It does cross-build correctly when using the conan center msys, like I stated above the reason for wanting to use a pre-installed version of msys is because when run through our CI system the msys2 installer seems to hang during every other run. I've been able to reproduce the hanging issue when running it manually as well but it seems to occur less often.

**Comment by memsharded** [comment]
Maybe it is that your msys2 is missing some pacman installed packages inside it? Conan recipe installs ``"packages": "base-devel,binutils,gcc",``

**Comment by memsharded** [comment]
It would be good to know why it is hanging, this msys2 package is massively used in ConanCenter Windows machine, and we haven't found anything. Some things to check:
- Antivirus software
- Windows indexing

 Folder exclusions should be done for both, or they can interfere

**Comment by wbehrens-on-gh** [comment]
> It would be good to know why it is hanging, this msys2 package is massively used in ConanCenter Windows machine, and we haven't found anything. Some things to check:
> 
> * Antivirus software
> * Windows indexing
> 
> Folder exclusions should be done for both, or they can interfere

When you say windows indexing are you talking about https://support.microsoft.com/en-us/windows/search-indexing-in-windows-10-faq-da061c83-af6b-095c-0f7a-4dfecda4d15a

**Comment by memsharded** [comment]
Typically in WIndows -> Settings -> Exclude folders from being indexed

**Comment by wbehrens-on-gh** [comment]
Removing indexing from that folder hasn't seemed to solve the problem. Stranger yet the hanging issue only sometimes occurs (ran two identical builds, one hanged the other completed).

**Comment by memsharded** [comment]
> Removing indexing from that folder hasn't seemed to solve the problem. Stranger yet the hanging issue only sometimes occurs (ran two identical builds, one hanged the other completed).

Yes, that looks pretty random, this is why I was thinking about possible external factors like Antivirus scanners, or other OS applications that could be randomly affecting.

**Comment by wbehrens-on-gh** [comment]
Disabling antivirus scanners on that folder isn't really an option sadly, that's part of why I was wondering if the way I was attempting to use a pre-installed subsystem was wrong as with the setup I have that would be much preferred anyways.

**Comment by memsharded** [comment]
> Disabling antivirus scanners on that folder isn't really an option sadly, that's part of why I was wondering if the way I was attempting to use a pre-installed subsystem was wrong as with the setup I have that would be much preferred anyway

I get it might not be desired as a permanent solution, but if it would be possible to do some temporal deactivation, just to run some checks, and try to rule out the effect of the AV, that would be good information anyway.

>  I was attempting to use a pre-installed subsystem was wrong as with the setup I have that would be much preferred anyways

Yeah, it should be possible to make the preinstalled msys2 work correctly, so lets keep trying. Things that would be possible to try:

- Check for possible differences in the msys2 provisioning. Maybe get the Conan package one, copy (deploy) the files, and then try to use it from your profile as you are doing above?
- Try to reduce the problem, and have some minimal case. Maybe this can be reproduced with an ``conan new autotools_lib`` template project instead of ``openssl``? Can we reproduce with a single command like ``conan install --requires=openssl/version ..``, to remove other potential factors
